sdk/lib/cxxrt/guard.cc

```diff
@@ -54,6 +56,8 @@ namespace
 
 		/**
 		 * Acquire the lock.
+		 *
+		 * This is safe only in IRQ-deferred context.
```

Comment:
Why?

Reply:
Oh, hmm.  This should be an atomic, I guess this predated support for atomics.  Unless there's a header somewhere defining these functions, it *isn't* called with interrupts disabled, so that would be a bug.

Reply:
The callers are given prototypes that mark them as IRQ deferring with `DEFINE_ATOMIC_LIBCALL` at the top of the file.

Reply:
Ah, right.  I wonder if we should (after this PR is merged) refactor this whole thing to use priority inheriting flag locks.  The lock half of the word is entirely ours to play with.